Source/Core/Core/Core.cpp

```diff
@@ -145,8 +152,8 @@ bool IsRunningInCurrentThread()
 }
 
 bool IsCPUThread()
-{
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+{	
+	return s_cpu_thread.joinable() ? tls_is_cpu_thread : !s_is_started;
```

Comment:
What exactly is s_is_started tracking? And should we just move setting it to directly above launching the CPU thread?